### PR TITLE
spring-hw-6

### PR DESCRIPTION
**Pull request by ShemsheyMaxim**

### CODE REVIEW

File: src/main/java/com/cinema/controllers/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    public AuthenticationController(UserMapper userMapper,
+                                    AuthenticationService authenticationService) {
+        this.userMapper = userMapper;
+        this.authenticationService = authenticationService;
+    }
+
+    @PostMapping("/register")
+    public void register(@RequestBody UserRequestDto userRequestDto) {
+        User user = userMapper.toUserEntity(userRequestDto);
+        authenticationService.register(user.getEmail(), user.getPassword());
```
Comment (MykolaFatkullin) [inline src/main/java/com/cinema/controllers/AuthenticationController.java]: You don't need to create a user object here, just use get methods from your Dto

File: src/main/java/com/cinema/model/dto/OrderRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.cinema.model.dto;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class OrderRequestDto {
+    private Long userId;
```
Comment (MykolaFatkullin) [inline src/main/java/com/cinema/model/dto/OrderRequestDto.java]: You don't need this, because you request userId from the URL

**Review by MykolaFatkullin — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/controllers/AuthenticationController.java (hunk 1 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/cinema/controllers/AuthenticationController.java]: done

File: src/main/java/com/cinema/model/dto/OrderRequestDto.java (hunk 2 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/cinema/model/dto/OrderRequestDto.java]: done

**Review by MykolaFatkullin — APPROVED** [review]

File: src/main/java/com/cinema/controllers/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderMapper orderMapper;
+    private final OrderService orderService;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+
+    public OrderController(OrderMapper orderMapper, OrderService orderService,
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/controllers/OrderController.java:27]: let's use the annotation above the constructor everywhere

File: src/main/java/com/cinema/model/dto/ShoppingCartRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package com.cinema.model.dto;
+
+import java.util.List;
+
+public class ShoppingCartRequestDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/ShoppingCartRequestDto.java]: do we use this dto?

File: src/main/java/com/cinema/model/dto/ShoppingCartResponseDto.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package com.cinema.model.dto;
+
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Long> ticketIds;
+    private Long userId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/ShoppingCartResponseDto.java]: see checklist saying about ShoppingCartResponseDto

File: src/main/java/com/cinema/service/mapper/impl/OrderMapperImpl.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+@Component
+public class OrderMapperImpl implements OrderMapper {
+    @Override
+    public OrderResponseDto toOrderDto(Order order) {
+        OrderResponseDto responseDto = new OrderResponseDto();
+        responseDto.setId(order.getId());
+        responseDto.setUserId(order.getUser().getId());
+        responseDto.setOrderDate(order.getOrderDate());
+        List<Long> ticketsId = order.getTickets()
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/mapper/impl/OrderMapperImpl.java]:
```suggestion
        List<Long> ticketIds = order.getTickets()
```

File: src/main/java/com/cinema/service/mapper/impl/ShoppingCartMapperImpl.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        List<Long> ticketsId = shoppingCart.getTickets()
+                .stream()
+                .map(Ticket::getId)
+                .collect(Collectors.toList());
+        responseDto.setTicketIds(ticketsId);
+        return responseDto;
+    }
+
+    @Override
+    public ShoppingCart toShoppingCartEntity(ShoppingCartRequestDto shoppingCartRequestDto) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/mapper/impl/ShoppingCartMapperImpl.java]: do we need this method?

File: src/main/java/com/cinema/service/mapper/impl/ShoppingCartMapperImpl.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+                                  ShoppingCartService shoppingCartService) {
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+    }
+
+    @Override
+    public ShoppingCartResponseDto toShoppingCartDto(ShoppingCart shoppingCart) {
+        ShoppingCartResponseDto responseDto = new ShoppingCartResponseDto();
+        responseDto.setUserId(shoppingCart.getId());
+        List<Long> ticketsId = shoppingCart.getTickets()
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/mapper/impl/ShoppingCartMapperImpl.java]:
```suggestion
        List<Long> ticketIds = shoppingCart.getTickets()
```

File: src/main/java/com/cinema/service/mapper/impl/UserMapperImpl.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    @Override
+    public UserResponseDto toUserDto(User user) {
+        UserResponseDto userResponseDto = new UserResponseDto();
+        userResponseDto.setId(user.getId());
+        userResponseDto.setEmail(user.getEmail());
+        return userResponseDto;
+    }
+
+    @Override
+    public User toUserEntity(UserRequestDto userRequestDto) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/mapper/impl/UserMapperImpl.java]: do we need this method?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/controllers/OrderController.java (hunk 3 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/cinema/controllers/OrderController.java:27]: done

File: src/main/java/com/cinema/model/dto/ShoppingCartRequestDto.java (hunk 4 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/cinema/model/dto/ShoppingCartRequestDto.java]: fixed

File: src/main/java/com/cinema/model/dto/ShoppingCartResponseDto.java (hunk 5 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/cinema/model/dto/ShoppingCartResponseDto.java]: fixed

File: src/main/java/com/cinema/service/mapper/impl/OrderMapperImpl.java (hunk 6 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/cinema/service/mapper/impl/OrderMapperImpl.java]: done

File: src/main/java/com/cinema/service/mapper/impl/ShoppingCartMapperImpl.java (hunk 7 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/cinema/service/mapper/impl/ShoppingCartMapperImpl.java]: fixed

File: src/main/java/com/cinema/service/mapper/impl/ShoppingCartMapperImpl.java (hunk 8 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/cinema/service/mapper/impl/ShoppingCartMapperImpl.java]: done

File: src/main/java/com/cinema/service/mapper/impl/UserMapperImpl.java (hunk 9 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/cinema/service/mapper/impl/UserMapperImpl.java]: fixed

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**